Xamarin.Essentials/Launcher/Launcher.android.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System.Linq;
+using System.Threading.Tasks;
+using Android.Content;
+using Android.Content.PM;
+using AndroidUri = Android.Net.Uri;
+using Uri = System.Uri;
+
+namespace Xamarin.Essentials
+{
+    public static partial class Launcher
+    {
+        static Task<bool> PlatformCanOpenAsync(Uri uri)
+        {
+            var intent = CreateIntent(uri.ToString());
+            var manager = Platform.AppContext.PackageManager;
+            var supportedResolvedInfos = manager.QueryIntentActivities(intent, PackageInfoFlags.MatchDefaultOnly);
+            return Task.FromResult(supportedResolvedInfos.Any());
+        }
+
+        static Task PlatformOpenAsync(Uri uri)
+        {
+            var intent = CreateIntent(uri.ToString());
+            Platform.AppContext.StartActivity(intent);
```

Comment:
How will this work with the intent system? Does the top need to be cleared?

Reply:
The application is launched on top of the current application. The back button returns to the calling app. I don't think the top has to be cleared, it works well on android.